test/tools/base64_test.cpp

```diff
@@ -21,39 +21,275 @@
+    ASSERT_EQ(decodedValStr.data(), std::string(origStr));
+};
+
+/**
+* Test Case: Tests Base64 encode and decode when passing invalid data length
+* invalid length: 'invalid length > actual input array length'
+*/
+TEST_F(Base64EncodeDecodeTests, invalid_encoded_decoded_overFlowDatalength){
+   
+    static const char origStr[] = "abcdefg";
+    size_t origStrLen = sizeof(origStr);
+    size_t origStrInvalidLen = 15;
+    ASSERT_NE(origStrLen, origStrInvalidLen);
+
+    std::string encodedStrVal = uprotocol::tools::Base64::base64encode((uint8_t *)origStr,
+                                                                origStrLen);
+    std::string encodedStrInVal = uprotocol::tools::Base64::base64encode((uint8_t *)origStr,
+                                                                origStrInvalidLen);
+    ASSERT_NE(encodedStrVal, encodedStrInVal);
+
+    size_t encodedStrInValLen = 15;
+    std::string encodedStr = uprotocol::tools::Base64::base64encode(origStr);
+    std::string decodedValStr = uprotocol::tools::Base64::base64decode((uint8_t *)encodedStrVal.data(),
+                                                                    encodedStrVal.length());
+    std::string decodedInValStr = uprotocol::tools::Base64::base64decode((uint8_t *)encodedStrVal.data(),
+                                                                       encodedStrInValLen);
+                                                                    
+    ASSERT_NE(decodedValStr, decodedInValStr);
+    ASSERT_EQ(decodedValStr.data(), std::string(origStr));
+};
+
+/**
+* Test Case: Tests base64 encode and decode when passing null data
+* Throw error in the log "Invalid data"
+*/
+TEST_F(Base64EncodeDecodeTests, null_data){
    
-}
\ No newline at end of file
+    static const uint8_t* origStr = nullptr;
+    size_t origStrLen = sizeof(origStr);
+
+    std::string encodedStrVal = uprotocol::tools::Base64::base64encode(origStr,
+                                                                       origStrLen);
+    ASSERT_EQ(encodedStrVal, "");
+
+    std::string decodedValStr = uprotocol::tools::Base64::base64decode(origStr,
+                                                                       encodedStrVal.length());
+    ASSERT_EQ(decodedValStr, "");
+};
+
+/**
+* Test Case: Tests base64 encode and decode when passing lengh zero
+* Throw error in the log "Invalid data"
+*/
+TEST_F(Base64EncodeDecodeTests, length_0){
+   
+    static const char origStr[] = "abcdefg";
+    size_t origStrLen = 0;
+
+    std::string encodedStrVal = uprotocol::tools::Base64::base64encode((uint8_t *)origStr,
+                                                                       origStrLen);
+    ASSERT_EQ(encodedStrVal, "");
+
+    std::string decodedValStr = uprotocol::tools::Base64::base64decode((uint8_t *)encodedStrVal.data(),
+                                                                       encodedStrVal.length());
+    ASSERT_EQ(encodedStrVal, "");
```

Comment:
Validate decodedValStr, instead of encodedStrVal

Reply:
Fixed